src/EventStore.Core.Tests/Services/Transport/Tcp/TcpConnectionTests.cs

```diff
@@ -37,16 +38,15 @@ public class TcpConnectionTests {
 					new TcpClientConnector(),
 					TimeSpan.FromSeconds(5),
 					(conn) => mre.Set(),
-					(conn, error) => {
-						Assert.Fail($"Connection failed: {error}");
-					},
+					(conn, error) => failedError = error,
 					false);
 
 				var serverSocket = listeningSocket.Accept();
 				var serverTcpConnection = TcpConnection.CreateAcceptedTcpConnection(Guid.NewGuid(),
 					(IPEndPoint)serverSocket.RemoteEndPoint, serverSocket, false);
 
 				mre.Wait(TimeSpan.FromSeconds(3));
+				Assert.Null(failedError);
```

Comment:
since this is set on another thread it'll need some synchronisation otherwise the asserting thread might still have the null from above. a taskcompletion source with RunContinuationsAsynchronously  would do the trick i think

Reply:
Changed it.